libraries/tock-tbf/src/types.rs

```diff
@@ -304,6 +333,117 @@ impl core::convert::TryFrom<&[u8]> for TbfHeaderV2FixedAddresses {
     }
 }
 
+impl core::convert::TryFrom<&[u8]> for TbfHeaderDriverPermission {
+    type Error = TbfParseError;
+
+    fn try_from(b: &[u8]) -> Result<TbfHeaderDriverPermission, Self::Error> {
+        Ok(TbfHeaderDriverPermission {
+            driver_number: u32::from_le_bytes(
+                b.get(0..4)
+                    .ok_or(TbfParseError::InternalError)?
+                    .try_into()?,
+            ),
+            offset: u32::from_le_bytes(
+                b.get(4..8)
+                    .ok_or(TbfParseError::InternalError)?
+                    .try_into()?,
+            ),
+            allowed_commands: u64::from_le_bytes(
+                b.get(8..16)
+                    .ok_or(TbfParseError::InternalError)?
+                    .try_into()?,
+            ),
+        })
+    }
+}
+
+impl<const L: usize> core::convert::TryFrom<&[u8]> for TbfHeaderV2Permissions<L> {
+    type Error = TbfParseError;
+
+    fn try_from(b: &[u8]) -> Result<TbfHeaderV2Permissions<L>, Self::Error> {
+        let length = u16::from_le_bytes(
+            b.get(0..2)
+                .ok_or(TbfParseError::InternalError)?
```

Comment:
I don't think this is quite right. We can only return an internal error here if we are guaranteeing that b is at least 2 bytes long when this is called, so the only way that get() can fail is because the caller code changed. However, parse.rs is just passing in whatever is remaining, without doing a length check.

Reply:
Good catch! I have converted this to `TbfParseError::BadTlvEntry`